org.eclipse.m2e.core/src/org/eclipse/m2e/core/internal/project/registry/ProjectRegistryManager.java

```diff
@@ -974,13 +974,6 @@ MutableProjectRegistry newMutableProjectRegistry() {
    *           been created
    */
   void applyMutableProjectRegistry(MutableProjectRegistry newState, IProgressMonitor monitor) throws CoreException {
-    // don't cache maven sessions
-    for(MavenProjectFacade facade : newState.getProjects()) {
```

Comment:
I wonder if it is a problem if this is not done anymore?
Or is it irrelevant because clearProjectBuildingRequest() is called in the IMavenToolbox?

Reply:
Yes IMavenToolbox clears the reference now always.